src/cache.ts

```diff
@@ -69,7 +69,7 @@ export async function set(
 ): Promise<void> {
   const key = optionsToCacheKey(options);
 
-  const permissionsString = options.permissions
+  const permissionsString = isNonEmptyObject(options.permissions)
```

Comment:
I'm unclear in what circumstance `options.permissions` would be an empty object? It is either undefined, or whatever you set `options.permissions` to. When I run the current tests and log out `options.permissions`, it is never an empty object.

Could you add a test based on a real life usage that would fail without your change?

Reply:
As is stated by the warning in the README, if you use github enterprise <= 2.18, you have to supply an empty object for permissions when doing installation authentication. The hook calls the installation authentication path with a hardcoded empty options object, and thus undefined permissions. I'm merely trying to salvage compatibility with 2.18 with least intrusive change possible. I thought supplying an empty permissions object should be semantically identical to not specifying permissions at all, the tests however treated it differently.

Reply:
I can add a test, that shows the compatibility problems with 2.18. I.e. that it returns `permissions: undefined` if you don't include permissions. But maybe you're not willing to accept fixes for 2.18 since it's not supported anymore?

Reply:
We looked into this in the past and there were complications, which is why we ultimately decided to just add a warning to the README. 2.18 is now out of maintenance, so I'd rather not add code for it.

Thanks Andreas, but I think I'll go ahead and close the PR. I hope you'll be able to upgrade your GHE installation soon

Reply:
Okay, I'm sad to hear that... Out of curiosity though. Isn't it reasonable for `permissions: undefined` to mean the same thing as `permissions: {}`? Right now the cache logic will react differently in these circumstances...

Reply:
In theory yes. All I remember is that we tried that path but hit some roadblock which made that change more complicated than it seems. If you are interested, you can check the issues / pull requests around when we added the note about GHE 2.18 to the README.